src/Controllers/DatabaseController.js

```diff
@@ -198,7 +198,24 @@ const filterSensitiveData = (
   protectedFields,
   object
 ) => {
-  protectedFields && protectedFields.forEach(k => delete object[k]);
+  // Get all users with read access
+  if (protectedFields && protectedFields.length > 0) {
+    const userACLs = aclGroup.filter(acl => {
```

Comment:
Why don't you just go through the object ACLs and check if there is a rule with the user id and read true?

Reply:
As far as I understand the `aclGroup` contains all roles associated with the current user and the user id.  `aclGroup` is not the object's ACL. So I first use the `aclGroup` to retrieve the current user's id. Then I use the actual object to check if the user id has read permission using `getReadAccess`.

If there is a better way to retrieve the current user's id (instead of using the acl group) I'll gladly use it.

The `aclGroup` approach is used in the `addPointerPermissions` method too:
https://github.com/parse-community/parse-server/blob/ef14ca530d8fd1ceb3997697738f912a3fd8c288/src/Controllers/DatabaseController.js#L1467-L1482

 `Line 202` avoids the whole retrieval of the user id if there are no protected fields to remove.